Argobots: Missing _Atomic qualifier in threads_argobots_mutex.c

I tried building the Argobots integration with the current Clang trunk while testing #7675. Clang seems to be more strict about the `_Atomic` qualifier than GCC so the build fails: 

```
../../../../../opal/mca/threads/argobots/threads_argobots_mutex.c:107:13: error: address argument to atomic operation must be a pointer to _Atomic type ('intptr_t *' (aka 'long *') invalid)
        if (opal_atomic_compare_exchange_strong_ptr(
            ^
../../../../../opal/include/opal/sys/atomic_stdc.h:91:71: note: expanded from macro 'opal_atomic_compare_exchange_strong_ptr'
#define opal_atomic_compare_exchange_strong_ptr(addr, compare, value) atomic_compare_exchange_strong_explicit (addr, compare, value, memory_order_relaxed, memory_order_relaxed)
                                                                      ^                                        ~~~~
/home/h4/jschuch/opt/clang-trunk/lib/clang/11.0.0/include/stdatomic.h:127:49: note: expanded from macro 'atomic_compare_exchange_strong_explicit'
#define atomic_compare_exchange_strong_explicit __c11_atomic_compare_exchange_strong
                                                ^
../../../../../opal/mca/threads/argobots/threads_argobots_mutex.c:124:13: error: address argument to atomic operation must be a pointer to _Atomic type ('intptr_t *' (aka 'long *') invalid)
        if (opal_atomic_compare_exchange_strong_ptr((intptr_t *)cond,
            ^                                       ~~~~~~~~~~~~~~~~
../../../../../opal/include/opal/sys/atomic_stdc.h:91:71: note: expanded from macro 'opal_atomic_compare_exchange_strong_ptr'
#define opal_atomic_compare_exchange_strong_ptr(addr, compare, value) atomic_compare_exchange_strong_explicit (addr, compare, value, memory_order_relaxed, memory_order_relaxed)
                                                                      ^                                        ~~~~
/home/h4/jschuch/opt/clang-trunk/lib/clang/11.0.0/include/stdatomic.h:127:49: note: expanded from macro 'atomic_compare_exchange_strong_explicit'
#define atomic_compare_exchange_strong_explicit __c11_atomic_compare_exchange_strong
                                                ^
2 errors generated.
```

--- a/opal/mca/threads/argobots/threads_argobots_mutex.c
+++ b/opal/mca/threads/argobots/threads_argobots_mutex.c
@@ -105,7 +105,7 @@ void opal_mutex_create(struct opal_mutex_t *m)
         }
         void *null_ptr = OPAL_ABT_MUTEX_NULL;
         if (opal_atomic_compare_exchange_strong_ptr(
-             (intptr_t *)&m->m_lock_argobots, (intptr_t *)&null_ptr,
+             (opal_atomic_intptr_t *)&m->m_lock_argobots, (intptr_t *)&null_ptr,
              (intptr_t)abt_mutex)) {
             /* mutex is successfully created and substituted. */
             return;
@@ -121,7 +121,7 @@ static void opal_cond_create(opal_cond_t *cond)
         ABT_cond new_cond;
         ABT_cond_create(&new_cond);
         void *null_ptr = OPAL_ABT_COND_NULL;
-        if (opal_atomic_compare_exchange_strong_ptr((intptr_t *)cond,
+        if (opal_atomic_compare_exchange_strong_ptr((opal_atomic_intptr_t *)cond,
                                                     (intptr_t *)&null_ptr,
                                                     (intptr_t)new_cond)) {
             /* cond is successfully created and substituted. */

